File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque
def solution(priorities, location):
    answer = 0
    p=deque(priorities)
    while p:
        if p[0]==max(p):
            answer+=1
            p.popleft()
            if location==0:
                break
            else:
                location-=1
        else:
            p.append(p.popleft())
            if location==0:
                location+=(len(p)-1)
            else:
                location-=1
            
            
    
    return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;
import java.io.*;
class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        Deque<List<Integer>> deque = new ArrayDeque<>();
        PriorityQueue<Integer> pq = new PriorityQueue<>(Comparator.reverseOrder());
        for (int i=0; i<priorities.length; i++){
            deque.addLast(Arrays.asList(i,priorities[i]));
            pq.add(priorities[i]);
        }
        while (!deque.isEmpty()){
            List<Integer> process = deque.pollFirst();
            if (process.get(1)==pq.peek()){
                pq.poll();
                answer++;
                if (process.get(0) == location){
                    return answer;
                }
            }else{
                deque.addLast(process);
            }
        }
        return answer;
    }
}